cmd/feegrant.go

```diff
@@ -0,0 +1,288 @@
+				keyNameOrAddress string
+			)
+
+			if len(args) == 1 {
+				keyNameOrAddress = cl.Config.Key
+			} else {
+				keyNameOrAddress = args[1]
+			}
+
+			granterAddr, err = cl.AccountFromKeyOrAddress(keyNameOrAddress)
+
+			if err != nil {
+				return fmt.Errorf("did not find wallet %s", keyNameOrAddress)
+			}
+
+			granteeAddr, err = cl.DecodeBech32AccAddr(args[0])
+			if err != nil {
+				return err
+			}
+
+			grant, err := cl.QueryFeeGrant(granteeAddr, granterAddr)
+
+			if err != nil {
+				return err
+			}
+
+			return cl.PrintObject(grant)
+		},
+	}
+
+	return cmd
+}
+
+func feegrantFeeGrantCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "grant grantee",
+		Short: "Grant fee allowance to an address",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var (
+				granterAddr sdk.AccAddress
+				granteeAddr sdk.AccAddress
+				err         error
+			)
+
+			cl := config.GetDefaultClient()
+			key, _ := cmd.Flags().GetString(FlagFrom)
+			granterAddr, err = cl.AccountFromKeyOrAddress(key)
+			if err != nil {
+				return err
+			}
+
+			granteeAddr, err = cl.DecodeBech32AccAddr(args[0])
+			if err != nil {
+				return err
+			}
+
+			spendLimit, err := cmd.Flags().GetString(FlagSpendLimit)
+			if err != nil {
+				return err
+			}
+
+			spendLimitCoins, err := sdk.ParseCoinsNormalized(spendLimit)
+			if err != nil {
+				return err
+			}
+
+			exp, err := cmd.Flags().GetString(FlagExpiration)
+			if err != nil {
+				return err
+			}
+
+			basicGrant := feegrant.BasicAllowance{
+				SpendLimit: spendLimitCoins,
+			}
+
+			var expiresAtTime time.Time
+			if exp != "" {
+				expiresAtTime, err = time.Parse(time.RFC3339, exp)
+				if err != nil {
+					return err
+				}
+
+				basicGrant.Expiration = &expiresAtTime
+			}
+
+			var grant feegrant.FeeAllowanceI
+			grant = &basicGrant
+
+			periodClock, err := cmd.Flags().GetInt64(FlagPeriod)
+			if err != nil {
+				return err
+			}
+
+			periodLimitVal, err := cmd.Flags().GetString(FlagPeriodLimit)
+			if err != nil {
+				return err
+			}
+
+			if periodClock > 0 || periodLimitVal != "" {
+				periodLimit, err := sdk.ParseCoinsNormalized(periodLimitVal)
+				if err != nil {
+					return err
+				}
+
+				if periodClock <= 0 {
+					return fmt.Errorf("period clock was not set")
+				}
+
+				if periodLimit == nil {
+					return fmt.Errorf("period limit was not set")
+				}
+
+				periodReset := getPeriodReset(periodClock)
+				if exp != "" && periodReset.Sub(expiresAtTime) > 0 {
+					return fmt.Errorf("period (%d) cannot reset after expiration (%v)", periodClock, exp)
+				}
+
+				periodic := feegrant.PeriodicAllowance{
+					Basic:            basicGrant,
+					Period:           getPeriod(periodClock),
+					PeriodReset:      getPeriodReset(periodClock),
+					PeriodSpendLimit: periodLimit,
+					PeriodCanSpend:   periodLimit,
+				}
+
+				grant = &periodic
+			}
+
+			allowedMsgs, err := cmd.Flags().GetStringSlice(FlagAllowedMsgs)
```

Comment:
Should this be an argument too?

Reply:
why should optional args be arguments? seems it'd be complicated to parse them all separately